### ISSUE TITLE
Find/Replace dialogue changing strings on its own

### ISSUE THREAD

**Issue by HawardKetoyoMsatsi** [africanmathsinitiative/R-Instat#6720]
@rdstern @shadrackkibet @lloyddewit @dannyparsons In an attempt to fix #6534 I found Something strange happening with the String Handling dialog and the script window. 
I generated an R code in the script window from the dialog with one single quote ('). I later left without running anything. The script window later accumulated many other single quotes now the code is messy.

The code I created
![image](https://user-images.githubusercontent.com/16137042/132191406-cd013955-a2bb-4f79-b507-26cfef39c961.png)

The code updated itself after leaving R-instat for some time without running anything.
![image](https://user-images.githubusercontent.com/16137042/132191203-b8ab4df4-f747-4071-8437-fcaa79976a17.png)

**Comment by Patowhiz** [comment]
We are now looking into this with @HawardKetoyoMsatsi

**Comment by lloyddewit** [comment]
I tried to recreate this bug. The closest I could get is shown in the screenshot below. I think this bug no longer exists so I will close this issue.
@rdstern Please could you also try and recreate? And if successful reopen the issue?
Thanks!

![image](https://user-images.githubusercontent.com/57253949/197381045-6ff0c5d6-4217-423c-ae08-c2593dafb9cc.png)

**Comment by lloyddewit** [comment]
I managed to recreate the bug. I pasted the R code from PR #6619 into the script window. When I came back about an hour later, a line containing single quotes (sse below) was surrounded by many more single quotes. Lines with double quotes were unaffected.
Since this issue was opened, the Script window has been completely rewritten and now uses the third-party Scintilla control. Therefore, I doubt that the issue is caused by the script window code or an R-Instat dialog. I suspect someything is happening at the dotnet level. However, I searched online and couldn't find anyone else reporting the issue. So maybe it is an R-Instat problem?

I will reopen the issue.

```
EmergDate <- as.Date('2000/01/19')
```
... after 20 minutes:
```
EmergDate <- as.Date(''''2000/01/19'''')
```
The same happened with simpler lines that I added to test:
```
a <- ''''b''''
''''c''''
```

**Comment by Patowhiz** [comment]
@lloyddewit very strange indeed. I wonder which other R script is capable of causing this. We could then get a pattern that can help with further investigation.

**Comment by lloyddewit** [comment]
> @lloyddewit very strange indeed. I wonder which other R script is capable of causing this. We could then get a pattern that can help with further investigation.

@Patowhiz I think it's any script that contains single quotes, even a minimal script like:
```
'a'
```
I don't think that it's not caused by any dialog, the problem is purely in the script window. 
I am currently running an R-Instat instance in debug mode. I put breakpoints on all the handler functions that are called when the text changes. This may give me more info.

**Comment by Patowhiz** [comment]
@lloyddewit aha yes. That should explain it.
I didn't know we changed `DoNotTranslate` to `DoNotDelete`. Or is that a knew key phrase you introduced in the translation system?

**Comment by lloyddewit** [comment]
> I didn't know we changed `DoNotTranslate` to `DoNotDelete`. Or is that a knew key phrase you introduced in the translation system?

@Patowhiz You are correct! It should be `DoNotTranslate`. I fixed it in my comment above.

**Comment by lloyddewit** [comment]
Fixed in PR #7988
